tools/engine_tool/lib/src/commands/build_command.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:path/path.dart' as path;
+import 'package:process_runner/process_runner.dart';
+
+import '../build_utils.dart';
+
+import 'command.dart';
+
+const String _configFlag = 'config';
+
+// Some wrinkles I'd like to iron out:
+//  - There can be multiple GlobalBuild objects with the same name. How
+//    should we deal with these conflicts? "mac_clang_tidy" and
+//    "mac_host_engine" both define "host_debug".
+//  - Naming: Should BuildConfig be BuilderConfig and GlobalBuild be BuildConfig?
+//  - The BuildConfig database seems to be lacking things I expected:
+//    - There is no "android_debug_*" GlobalBuild in any of the builders that run on mac?
+//    - The gn flags specified in the GlobalBuilds don't reference rbe or goma? Should this be a local configuration that the user selects?
+//    - Need to better understand available targets per GlobalBuild. The lists seem slim.
+
+// TODO(johnmccutchan): List all available build targets and allow the user
+// to specify which one(s) we should build on the cli.
+
+// TODO(johnmccutchan): Can we show a progress indicator like 'running gn...'?
+
+/// The root 'build' command.
+final class BuildCommand extends CommandBase {
+  /// Constructs the 'query' command.
+  BuildCommand({
+    required super.environment,
+    required Map<String, BuildConfig> configs,
+  }) {
+    builds = runnableBuilds(environment, configs);
+    // Add options here that are common to all queries.
+    argParser.addOption(
+      _configFlag,
+      abbr: 'c',
+      defaultsTo: 'host_debug',
```

Comment:
Definitely a TODO (unless you feel _inspired_) but I have the perfect default for you here:

https://github.com/flutter/engine/tree/main/tools/pkg/engine_repo_tools

Specifically:

https://github.com/flutter/engine/blob/df0dc1fc06ca8c8ff67b736ffddb5d6a3d295687/tools/pkg/engine_repo_tools/lib/engine_repo_tools.dart#L156-L169

You can see a recent example of this being used here (not exactly the same, but close):

https://github.com/flutter/engine/blob/df0dc1fc06ca8c8ff67b736ffddb5d6a3d295687/testing/scenario_app/bin/android_integration_tests.dart#L44-L49

Reply:
Personally I'd rather have a deterministic default. Crank lets you set configs, like your default builder, in a `~/.config/crank/config.json` file. You can also add your own custom builder in this config file in case the CI builders don't have your desired scenario.

Reply:
Hardcoding a default for now sgtm. I suspect we'll want a config file with defaults at some point, but that's big enough for a separate PR.